src/Functions/parseTimeDelta.cpp

```diff
@@ -0,0 +1,243 @@
+        {
+            auto col_to = ColumnFloat64::create();
+            auto & res_data = col_to->getData();
+
+            for (size_t i = 0; i < input_rows_count; ++i)
+            {
+                std::string_view str{arguments[0].column->getDataAt(i)};
+                Int64 length = str.length();
+                Int64 fast = 0;
+                Int64 slow = 0;
+                Int64 last_pos = 0;
+                Float64 result = 0;
+
+                /// ignore '.' and ' ' at the end of string
+                for (Int64 j = length - 1; j >= 0; j--)
+                {
+                    if (str[j] != '.' && str[j] != ' ')
+                    {
+                        last_pos = j;
+                        break;
+                    }
+                }
+
+                /// no valid characters
+                if (last_pos == 0)
+                {
+                    res_data.emplace_back(0);
+                    continue;
+                }
+
+                while (fast <= last_pos)
+                {
+                    ScanIgnoreCharacters(str, fast, last_pos);
+                    slow = fast;
+
+                    /// scan unsigned integer
+                    if (!ScanUnsignedInteger(str, fast, last_pos))
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName() + " find number failed: " + String(str) + ".",
+                            ErrorCodes::BAD_ARGUMENTS);
+                    }
+
+                    /// if there is a '.', then scan another integer to get a float number
+                    if (fast <= last_pos && str[fast] == '.')
+                    {
+                        fast++;
+                        if (!ScanUnsignedInteger(str, fast, last_pos))
+                        {
+                            throw Exception(
+                                "bad auguments invalid expression for function " + getName()
+                                    + " find number after '.' failed: " + String(str) + ".",
+                                ErrorCodes::BAD_ARGUMENTS);
+                        }
+                    }
+
+                    /// convert float/integer string to float
+                    Float64 base = 0;
+                    std::string_view base_str = str.substr(slow, fast - slow);
+                    auto value = boost::convert<Float64>(base_str, boost::cnv::strtol());
+                    if (!value.has_value())
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName()
+                                + " convert string to float64 failed:  " + String(base_str) + ".",
+                            ErrorCodes::BAD_ARGUMENTS);
+                    }
+                    base = value.get();
+
+                    ScanIgnoreCharacters(str, fast, last_pos);
```

Comment:
Do we really want to allow all ignore characters between number and unit name?
Expressions like `2-: +,min` will be parsed, but it doesn't look like a valid expression, let's ignore only spaces here